standards/nextjs_standards.md

```diff
@@ -0,0 +1,95 @@
+- **Readability**: Write code that is easy to read and understand, with meaningful variable and function names.
+- **Simplicity**: Keep solutions simple and avoid unnecessary complexity.
+- **Performance**: Optimize for speed and efficiency, considering the performance implications of your code.
+
+## Writing
+
+- Write clear, concise sentences and avoid tangents.
+- Use simple words; for example, use "use" instead of "utilize".
+- Use an active voice for readability.
+- Avoid subjective words like "easy", "quick", "simple", "just", etc.
+- Write in the second person (you/your) and use gender-neutral language.
+- Ensure code examples are properly formatted and working2.
+
+## Code Formatting
+
+- Use Prettier for automatic code formatting to maintain consistency.
+- Configure ESLint with Next.js specific rules to catch common issues and enforce style guidelines.
+- Adopt a consistent indentation method (e.g., 2 spaces) across the project.
+
+## Naming Conventions
+
+- **Files**: Use PascalCase for files (e.g., `MyComponent.ts`).
+- **Directories**: Use kebab-case for directories (e.g., `my-directory`).
+- **Components**: Use PascalCase for React components (e.g., `UserProfile.js`).
+- **Variables and Functions**: Use camelCase for variables and functions (e.g., `getUserData`).
+
+## File Structure & Naming Convention
+
+- Organize project files under the `src` directory for clarity.
+- Use the `pages` directory for page components, leveraging Next.js's file-based routing.
+- Store static assets in the `public` directory.
+- Group components into a `components` directory, categorizing further if necessary.
+
+## Component Structure
+
+Use atomic design when organizing components. Organize components into atoms, molecules, organisms, and templates for scalability and maintainability.
+
+## Efficient Code Splitting, Dynamic Imports, and Loading of Client-Side Scripts
+
+- Leverage Next.js's built-in code splitting to manage application code in smaller, more manageable units.
+- Utilize dynamic imports for loading non-essential components on demand, reducing the initial bundle size and improving page load times.
+- Identify components or code segments not critical for the initial page load and load them asynchronously.
+
+## Rendering Modes
+
+- **Server-Side Rendering (SSR)**: Use for pages requiring dynamic content or SEO, ideal for public-facing and content-heavy pages.
+- **Static Site Generation (SSG)**: Use for content with infrequent changes, suitable for landing pages, blogs, and marketing sites.
+- **Client-Side Rendering (CSR)**: Use for interactive components and real-time data, prioritizing pages/components that require interactivity.
+- **Hybrid Rendering**: Combine SSR and CSR benefits for applications with mixed content and interactivity needs.
+
+## Styling
+
+- Utilize CSS Modules for component-level styles to avoid conflicts.
+- Consider CSS-in-JS solutions like styled-components for dynamic styling.
```

Comment:
We always prefer tailwindcss, let's remove these 2

Reply:
Ah yes, will update this